sale_by_packaging/models/sale_order_line.py

```diff
@@ -76,44 +76,29 @@ def _force_qty_with_package(self):
     def _onchange_product_uom_qty(self):
         self._force_qty_with_package()
         res = super()._onchange_product_uom_qty()
-        is_pack_multiple_warning = self._check_qty_is_pack_multiple()
-        if is_pack_multiple_warning:
-            self.product_packaging_qty = False
-            self.product_packaging = False
-        return res if res else is_pack_multiple_warning
-
-    def _check_qty_is_pack_multiple(self):
-        """ Check only for product with sell_only_by_packaging
-        """
-        # and we dont want to have this warning when we had the product
-        if self.product_id.sell_only_by_packaging:
-            if not self._get_product_packaging_having_multiple_qty(
-                self.product_id, self.product_uom_qty, self.product_uom
-            ):
-                warning_msg = {
-                    "title": _("Product quantity cannot be packed"),
-                    "message": _(
-                        "For the product {prod}\n"
-                        "The quantity {qty} is not the multiple of any packaging.\n"
-                        "Please add a packaging."
-                    ).format(prod=self.product_id.name, qty=self.product_uom_qty),
-                }
-                return {"warning": warning_msg}
-        return {}
+        return res
 
     def _get_product_packaging_having_multiple_qty(self, product, qty, uom):
         if uom != product.uom_id:
             qty = uom._compute_quantity(qty, product.uom_id)
         return product.get_first_packaging_with_multiple_qty(qty)
 
+    def _inverse_product_packaging_qty(self):
+        # Force skipping of auto assign
+        # if we are writing the product_uom_qty directly via inverse
+        super(
+            SaleOrderLine, self.with_context(_skip_auto_assign=True)
+        )._inverse_product_packaging_qty()
+
     def write(self, vals):
         """Auto assign packaging if needed"""
-        fields_to_check = ["product_id", "product_uom_qty", "product_uom"]
-        if vals.get("product_packaging") or not any(
-            fname in vals for fname in fields_to_check
-        ):
+        if vals.get("product_packaging") or self.env.context.get("_skip_auto_assign"):
```

Comment:
```suggestion
        if "product_packaging" in vals.keys() or self.env.context.get("_skip_auto_assign"):
```
The product_packaging value can be False.

Reply:
the branch is on camptocamp: you can push to it. Or, merge it and change what you want in your branch :wink:

Reply:
But what is the reason behind those changes ?
Because they break 4 out of 7 tests
Where they included before ?